ingestion/src/kldata.rs

```diff
@@ -18,17 +19,12 @@ pub struct ObsinnChunk<'a> {
     observations: Vec<ObsinnObs<'a>>,
```

Comment:
General question about the term 'obsinn'. Does this mean that the dependency between lard and Obsinn goes both ways, i.e. that also lard needs to know about obsinn-specific formats and names? While this is probably fine, I'm just wondering if the design will also support other input sources than Obsinn. Will those input sources then have to mimic Obsinn somehow to be ingestible by lard?

Reply:
The ingestor is designed to be as source-indifferent as possible.

We have generic types like `Datum` and `DataChunk`, and some functions that can QC and insert them into the DB.

To support a new data source, you do only the source specific things:

1. Define a way to receive data (i.e a new HTTP endpoint, a Kafka consumer, etc.).
2. Provide a parser for the format you receive (This is where source-specific types like `ObsinnObs` and `ObsinnChunk` may come in).
3. Provide some code to fetch or create source specific labels for the data.
4. Finish parsing by providing the data in the form of the generic types mentioned above.

In the file you're commenting on you are seeing Obsinn-specific types because that file contains the parser for Obsinn's message format "kldata".

Reply:
Design makes sense to me. I actually have a relevant case. For this frost-ingest experiment I'm working on (designed for letting e.g. Avinor upload sets of observations as XML via the /put endpoint in a Frost instance), one option is that Frost passes those observations (decoded from the XML data) directly to lard via the lard ingestor. Will I then have two options: 1) format the data exactly like Obsinn does (so that the ingestor can't tell the difference), or 2) extend the ingestor with a parser for a new format? And then it boils down to who will be doing the translation between Avinor names and MET names; who will decide what type IDs and param IDs to use etc. Do you have any opinions on whether 1) or 2) would be the best choice in this case?

Reply:
That can work. Whether 1 or 2 is best will depend on the format you receive the data in but most probably 2 is best.

Make an issue to track this, and we can discuss in more detail tomorrow.

Reply:
Made [issue](https://github.com/metno/lard/issues/59).